ion-schema/src/isl/isl_type.rs

```diff
@@ -55,11 +59,22 @@ impl IslType {
 pub struct IslTypeImpl {
     name: Option<String>,
     constraints: Vec<IslConstraint>,
+    // Represents the ISL type struct in string format for anonymous type definition
+    // For named type definition, this will be `None`
+    pub(crate) isl_type_struct: Option<String>,
 }
 
 impl IslTypeImpl {
-    pub fn new(name: Option<String>, constraints: Vec<IslConstraint>) -> Self {
-        Self { name, constraints }
+    pub fn new(
+        name: Option<String>,
+        constraints: Vec<IslConstraint>,
+        isl_type_struct: Option<String>,
```

Comment:
What's the rationale for using `Option<String>` instead of a reference to the `Element`? (`Option<Weak<Element>>` maybe?)